File: cpp/leetcode/243.cpp
class Solution {
public:
    int shortestDistance(vector<string>& words, string word1, string word2) {
        int idx1 = -1, idx2 = -1, res = words.size() + 1;
        for(int i = 0; i < words.size(); i++){
            if(words[i] == word1) idx1 = i;
            else if(words[i] == word2) idx2 = i;
            if(idx1 != -1 && idx2 != -1) res = min(abs(idx2 - idx1), res);
        }
        return res;
    }
};

File: python/leetcode/243.py
class Solution:
    def shortestDistance(words, word1, word2):
        n, idx1, idx2, dist = len(words), -1, -1, 2**31
        for i in xrange(n):
            if words[i] == word1: idx1 = i
            elif words[i] == word2: idx2 = i
            if idx1 != -1 and idx2 != -1:
                dist = min(dist, abs(idx1 - idx2))
        return dist